### PR TITLE
WAC-42: Improve datapusher deployment.

### PR DESCRIPTION
**Pull request by Kacpersoftberry**
WAC-42: Improve datapusher deployment.

## Description

Improve deployment process of datapusher. Add comment for common error.

## Checklist

Put an `x` in the boxes that apply to this pull request (you can also fill these out after opening the pull request).
You may not need to check all boxes.

- [ ] The Jira ticket for this issue has been updated to "Ready to Review" or equivalent.
- [ ] I have developed these changes in discussion with the appropriate project manager.
- [ ] My code follows the general Fjelltopp documentation (see Confluence).
- [ ] I have made corresponding changes to the Fjelltopp documentation (see Confluence).
- [x] I have rebased this branch with master.
- [ ] New dependency changes have been committed.
- [ ] I have added automated tests that prove my fix is effective or that my feature works.
- [x] New and existing tests pass locally with my changes.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own code.
- [x] I have assigned at least one reviewer.
- [x] I have assigned at least one label to this PR: "patch", "minor", "major".

### CODE REVIEW

File: roles/ckan/tasks/deploy.yml [hunk 1]
```diff
@@ -81,12 +81,14 @@
...
     login_password: "{{ ckan_postgres_password }}"
-  when: (fjelltopp_env_type == 'local') and (ckan_datapusher_enable)
+  when: (fjelltopp_env_type == 'local') and (ckan_datapusher_enable == 'true')
   register: db_ping_result
   retries: 10
   delay: 5
   until: db_ping_result.is_available
   failed_when: not db_ping_result.is_available
+  # In case this job fails cause there is no object db_ping_result.is_available, check if you
+  # are on proper virtualenv or install psycopg2 python package
```
Comment (jonathansberry) [inline roles/ckan/tasks/deploy.yml]: Sadly this still doesn't work for me. There has been many hours now spent on this from across the team.  I think we have two options: get on a call asap and try to solve this in a screen-share, or we just delete the db wait step for now because it is interrupting development too much. Happy to accept another PR at some point with a fixed feature to test.

File: roles/ckan/templates/ckan/ckan_production.ini [hunk 2]
```diff
@@ -195,6 +195,7 @@ ckan.datapusher.url = {{ ckan_datapusher_url }}
 {% if ckan_datapusher_api_token %}
 ckan.datapusher.api_token = {{ ckan_datapusher_api_token }}
 {% endif %}
+ckan.datapusher.callback_url_base = http://ckan:5000/
```
Comment (jonathansberry) [inline roles/ckan/templates/ckan/ckan_production.ini:198]: Unfortunately I am still getting this error when uploading files to the datastore: 
image

Is that related to this setting?

**Review by jonathansberry — CHANGES_REQUESTED** [review]
Thanks Kacper, unfortunately things are still not working for me.  Happy to get on a screenshare if you think that is helpful.

File: roles/ckan/templates/ckan/ckan_production.ini (hunk 2 above)
Comment (Kacpersoftberry) [reply roles/ckan/templates/ckan/ckan_production.ini:198]: I need to add also documentation on confluence about setting up permissions to datastore. You need to login to ckan pod and run command `ckan datastore set-permissions` and copy output of that. After that open db pod, login to database and execute output of the previous command. More about it here: https://docs.ckan.org/en/2.9/maintaining/datastore.html#set-up-the-database under section "Set permissions".

PS. We can't do this over ssh cause we don't have a user with proper UID 🥲

**Review by jonathansberry — APPROVED** [review]
